types/src/util/tag.rs

```diff
@@ -36,6 +37,11 @@ impl Tag {
             return Err(HashingError::InvalidFilter)
         }
 
+        // Check if the filter has trailing zero
+        if bitvec[0] == 0 {
```

Comment:
It seems that you need to handle zero-length `bitvec`.

```
--- tests::executor_tests::chk_sig::sign_single_input_all_output stdout ----
thread 'tests::executor_tests::chk_sig::sign_single_input_all_output' panicked at 'index out of bounds: the len is 0 but the index is 0', /Users/travis/build/rust-lang/rust/src/libcore/slice/mod.rs:2079:10
```

Reply:
I fixed it and added a test case for this.